### DISCUSSION TITLE
Give users the option to dynamically provide Excel sheet names when needed.

### DISCUSSION THREAD

**Discussion by saiprakashpudi** [sskorol/test-data-supplier#196] [Q&A] [answered]
Fetching data from various sheets based on sheet names dynamically will be helpful for users who want to get data from multiple Excel sheets.

**Comment by sskorol** [comment]
@saiprakashpudi can you elaborate on the expected output? Each sheet may represent a completely different entity type. In this case, you can't put these types into a list unless all of them have a generic parent. And if I put them into a list of `Object`, you won't be able to differentiate them afterwards. On the other hand, if all the sheets have the same structure, then I can provide a chain method for an array of sheet names. And then, you'll get a list of objects of a given type. Would it work for you? If yes, can you create a feature request rather than discussion thread?

**Reply by saiprakashpudi** [reply]
Thanks for the quick response. Yes. It is useful when all the sheets have the same structure. So that we can provide the sheet name at runtime.

**Reply by sskorol** [reply, marked as answer]
Implemented in #198
